src/main/java/net/sf/jabref/logic/util/strings/StringUtil.java

```diff
@@ -167,7 +167,7 @@ public static String wrap(String in, int wrapAmount) {
 
         String[] lines = in.split("\n");
         StringBuilder result = new StringBuilder();
-        addWrappedLine(result, lines[0], wrapAmount);
+        addWrappedLine(result, lines[0].replaceAll("\\s+$", ""), wrapAmount);
```

Comment:
After reading the code a few times, I think that if the preference "Line wrap as in 2.9.2" is removed, so is this function. May be worth double-checking a few times though... :-)